server_cmds.go

```diff
@@ -340,14 +340,125 @@ func (s *ServerMethods) PrivilegeKeyAdd(ttype, id1, id2 int, options ...CmdArg)
+	// Takes the form of "/channelID"
+	DefaultChannel   string `ms:"client_default_channel"`
+	DefaultToken     string `ms:"client_default_token"`
+	Description      string `ms:"client_description"`
+	LoginName        string `ms:"client_login_name"`
+	NicknamePhonetic string `ms:"client_nickname_phonetic"`
+	SecurityHash     string `ms:"client_security_hash"`
+	TalkRequestMsg   string `ms:"client_talk_request_msg"`
+	UniqueIdentifier string `ms:"client_unique_identifier"`
+	FlagAvatar       string `ms:"client_flag_avatar"`
+	Platform         string `ms:"client_platform"`
+	// Which version of the Teamspeak client application this client uses.
+	Version     string `ms:"client_version"`
+	VersionSign string `ms:"client_version_sign"`
+
+	// Milliseconds since the client connected to the server.
+	ConnectionConnectedTime        int64 `ms:"connection_connected_time"`
+	ConnectionBytesReceivedTotal   int64 `ms:"connection_bytes_received_total"`
+	ConnectionBytesSentTotal       int64 `ms:"connection_bytes_sent_total"`
+	ConnectionPacketsReceivedTotal int64 `ms:"connection_packets_received_total"`
+	ConnectionPacketsSentTotal     int64 `ms:"connection_packets_sent_total"`
+	// Milliseconds since the client did something, for example sending
+	// a message, muting themselves or talking.
+	IdleTime             int64 `ms:"client_idle_time"`
+	MonthBytesDownloaded int64 `ms:"client_month_bytes_downloaded"`
+	MonthBytesUploaded   int64 `ms:"client_month_bytes_uploaded"`
+	TotalBytesDownloaded int64 `ms:"client_total_bytes_downloaded"`
+	TotalBytesUploaded   int64 `ms:"client_total_bytes_uploaded"`
+
+	// Current bandwidth used for outgoing file transfers (Bytes/s)
+	ConnectionFiletransferBandwidthSent int `ms:"connection_filetransfer_bandwidth_sent"`
+	// Current bandwidth used for incoming file transfers (Bytes/s)
+	ConnectionFiletransferBandwidthReceived int `ms:"connection_filetransfer_bandwidth_received"`
+	// Average bandwidth used for outgoing data in the last second (Bytes/s)
+	ConnectionBandwidthSentLastSecondTotal int `ms:"connection_bandwidth_sent_last_second_total"`
+	// Average bandwidth used for outgoing data in the last minute (Bytes/s)
+	ConnectionBandwidthSentLastMinuteTotal int `ms:"connection_bandwidth_sent_last_minute_total"`
+	// Average bandwidth used for incoming data in the last second (Bytes/s)
+	ConnectionBandwidthReceivedLastSecondTotal int `ms:"connection_bandwidth_received_last_second_total"`
+	// Average bandwidth used for incoming data in the last minute (Bytes/s)
+	ConnectionBandwidthReceivedLastMinuteTotal int `ms:"connection_bandwidth_received_last_minute_total"`
+	ChannelGroupID                             int `ms:"client_channel_group_id"`
+	ChannelGroupInheritedChannelID             int `ms:"client_channel_group_inherited_channel_id"`
+	NeededServerqueryViewPower                 int `ms:"client_needed_serverquery_view_power"`
+	Servergroups                               int `ms:"client_servergroups"`
```

Comment:
This is wrong, it should be an array of group id's.
The current decoder is unable to decode this properly even if you correct it though:
`* cannot parse 'client_servergroups[0]' as int: strconv.ParseInt: parsing "1,2,3": invalid syntax`

Reply:
I've written a hook that allows decoding to slices:

```go
// sliceHookFunc supports int/string decoding to slice
func sliceHookFunc(from reflect.Kind, to reflect.Kind, data interface{}) (interface{}, error) {
	if to == reflect.Slice {
		if from == reflect.String {
			return strings.Split(data.(string), ","), nil
		}

		if from == reflect.Int {
			return []int{data.(int)}, nil
		}
	}

	return data, nil
}
```

this would have to be passed to DecodeHook:

```go
DecodeHook: mapstructure.ComposeDecodeHookFunc(
    sliceHookFunc,
    timeHookFunc,
),
```

Reply:
Good catch! This has actually killed my teamspeak bot multiple times already.

Thanks for the code. I've created a commit with it and attributed you. [Github says here that you'd be credited automatically, but I couldn't find a button to take accept your code so I did it manually]